File: week4/implement-queue-using-stacks.java
class MyQueue {
    Stack<Integer> s1;
    Stack<Integer> s2;

    /** Initialize your data structure here. */
    public MyQueue() {
        s1 = new Stack<Integer>();
        s2 = new Stack<Integer>();
    }
    
    /** Push element x to the back of queue. */
    public void push(int x) {
        s1.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    public int pop() {
        // Move all items in s1 to s2
        while (!s1.isEmpty()) {
            s2.push(s1.pop());
        }
        int store = s2.pop();
        // Move everything back
        while (!s2.isEmpty()) {
            s1.push(s2.pop());
        }
        // Return last item moved to s2
        return store;
    }
    
    /** Get the front element. */
    public int peek() {
        // Move all items in s1 to s2
        while (!s1.isEmpty()) {
            s2.push(s1.pop());
        }
        int store = s2.peek();
        // Move everything back
        while (!s2.isEmpty()) {
            s1.push(s2.pop());
        }
        // Return last item moved to s2
        return store;
    }
    
    /** Returns whether the queue is empty. */
    public boolean empty() {
        return s1.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: summer2021/unit4/implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stack = []
        

    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.stack.append(x)

    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        stack2 = []
        while len(self.stack) > 1:
            stack2.append(self.stack.pop())
        poppedItem = self.stack.pop()
        while (len(stack2)) > 0:
            self.stack.append(stack2.pop())
        return poppedItem        

    def peek(self) -> int:
        """
        Get the front element.
        """
        stack2 = []
        while len(self.stack) > 1:
            stack2.append(self.stack.pop())
        poppedItem = self.stack[0]
        while (len(stack2)) > 0:
            self.stack.append(stack2.pop())
        return poppedItem    
        

    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        return len(self.stack) == 0


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()